### ISSUE TITLE
Reproducing Brant-2

### ISSUE THREAD

**Issue by MisterKloudy** [yzz673/Brant-2#1]
Hi, thank you very much for open-sourcing your code for this paper.
I am trying to figure out how to use the model for a new dataset.

It seems that the inputs to the model is in the format of data, psd and labels.
Could anyone share what is the expected format of the data and psd?

I've noticed that it is commented that the data should be of shape [batch size, channel number, 16, patch length] and psd should be of shape [batch size, channel number, 16, psd length].

I have a few questions on this:
(1) Does channel number refer to the number of channels, and can it be variable for a single dataset?
(2) What does the 16 refer to? My new dataset only has 2s recording of 200Hz so there's only 400 samples per observation and it seems that the model runs through too many conv blocks and hits an error. What is the minimum size of the input data?
(3) What is the recommended patch length?
(4) What is the psd length referring to? Is it the 256 frequencies calculated from the fft by default?

Thank you for your time!

**Comment by yzz673** [comment]
Thank you for your interest in our work. The answers of your questions are as follows:
(1) Yes, the channel number refer to the number of channels. If your task involves pooling the representations of different channels (i.e., your data is of shape [batch size, channel number, 16, patch length] and your label is of shape [batch size]), then the channel number cannot change within a single dataset. Else (i.e., your data is of shape [batch size, channel number, 16, patch length] and your label is of shape [batch size, channel number]), the channel number can be variable for a single dataset.
(2) 16 refers to the number of patches of a data sequence. In our experiments, we tried a minimum of patch length of 64 (i.e. 64*16=1024 samples per observation). For your case, one possible solution is interpolating your data upward.
(3) The model is pre-trained with patch length of 256, 512, 1024 and 2048. These patch lengths may result in better performance. But we experiment on patch length of 64 and 128 which alse work well.
(4) For each patch, we calculate its psd. The psd length refers to resulting length (i.e. patch length//2+1). The 256 frequencies calculated from the fft by default.

**Comment by MisterKloudy** [comment]
Hi, thank you very much for the answers!
A follow-up clarification on the psd: If the psd is of shape [batch size, channel number, 16, psd length], and I use the minimum patch length of 64, is the psd length = patch length //2 + 1 (i.e. 33) or is it psd length = 256?

**Comment by yzz673** [comment]
psd length=33
